compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -61,7 +61,12 @@ impl<'a> Parser<'a> {
                 },
                 _ => false,
             },
-            NonterminalKind::Path | NonterminalKind::Meta => match &token.kind {
+            NonterminalKind::Meta => match &token.kind {
+                token::PathSep | token::Ident(..) => true,
+                token::Interpolated(nt) => may_be_ident(&nt.0),
+                _ => token.is_keyword(kw::Unsafe),
```

Comment:
Isn't the `token.is_keyword(kw::Unsafe)` case covered by the first match arm via `token::Ident(..)`? I don't see a separate `TokenKind` for keywords.